### ISSUE TITLE
add wiki page about file structure

### ISSUE THREAD

**Issue by Eoksni** [emotz/guidelines#2]

**Comment by Eoksni** [comment]
https://github.com/emotz/guidelines/wiki/File-structure

**Comment by Eoksni** [comment]
@ITarako глянь может есть что дополнить

**Comment by ITarako** [comment]
> [directories.doc](https://docs.npmjs.com/files/package.json#directoriesdoc)
> 
> Put markdown files in here. Eventually, these will be displayed nicely, maybe, someday.

Это все таки для документации, а не для макетов и т.п.

**Comment by Eoksni** [comment]
макет это часть тз ( тех задание ), а тз это документ. а документу в документации - самое то. так что все ок :)

не знаю куда эти макеты еще пихать, по моему в доке им вполне комфортно. предложи решение если есть идеи

**Comment by ITarako** [comment]
видел вариант с директорией `template`, но это совсем будет ад х)

и надо наверное указать для какого типа проектов, эта структура

**Comment by Eoksni** [comment]
нууу можно и template, но помоему это уже черезчур тщательное разделение

а что значит "для какого типа проектов" ? для всех :)

**Comment by ITarako** [comment]
подумать над директорией для темлейтов

**Comment by Eoksni** [comment]
А собственно, что тут думать.
`frontend/src/templates`
м?

**Comment by Eoksni** [comment]
Зато я вспомнил про картинки. 

Во первых, статичные картинки для сайта, вроде логотипа и т.д. Их я думаю в `/frontend/src/static`. 

Во вторых, ресурсы, загружаемые пользователем. Напр. картинки в случае димановского инстаграмма. Их запилнуть в `/assets`.

**Comment by Eoksni** [comment]
Хотя картинки сложно считать исходниками. Наверное, лучше все таки `/frontend/static`.

**Comment by ITarako** [comment]
но тот же экспресс относит к статике *.html, *.css

**Comment by Eoksni** [comment]
его проблемы

**Comment by Eoksni** [comment]
а если более развернуто, то для экспресса статик - это то, что не требует обработки, что можно сервить юзеру "как-есть". в нашем случае css и html требуют обработки в общем случае на этапе компиляции.

**Comment by Eoksni** [comment]
в нашем случае экспрессовский статик - это папка `dist`

**Comment by ITarako** [comment]
ок, понял
вообще мне нравится использовать эту структуру (хотя возможно впринципе использование структуры ^^ )
поначалу было непривычно и слегка устрашал масштаб. но сейчас я точно знаю где что лежит, не бегая глазами по дереву проекта

**Comment by Eoksni** [comment]
Да, когда оно все разложено по полочкам это всегда лучше чем адовый хаос :) зацени я на вики замутил мега арт для отображения этой структуры.

**Comment by ITarako** [comment]
чотко, осталось как-то пацанов на это дело принудить х)

**Comment by Eoksni** [comment]
добавил `/backend/migrations` для db миграций

**Comment by ITarako** [comment]
`/backend/migrations/001_init_db.js` вместо `domain/src/db.async.proxy.js` или как

**Comment by Eoksni** [comment]
нет, твой db.async это модуль для работы с базой

а init_db.js это скрипт, инициализирующий базу (создающий таблицы, поля, триггеры). Для монго это тебе пока не важно, он schemeless, но для SQL баз миграции важны. Да и для монго потом тоже понадобятся.

**Comment by Eoksni** [comment]
а ведь при такой схеме
```
frontend
├───webpack.config.js
└───.eslintrc.json
```
получается, что `webpack.config.js` попадает под действие `.eslintrc.json`, хотя на самом деле тот для фронтенд проекта. Нехорошо. Напр., в еслинт конфиге для фронтенда нужно определять `env: browser` и т.д., которые для webpack.config.js не подходят.

**Comment by Eoksni** [comment]
Сдвинул .eslintrc.json в src

**Comment by Eoksni** [comment]
Добавил пункт насчет еслинтигнора в соответствии с https://github.com/emotz/kladovka/issues/38
